sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/PodSpecsCannotChangeNetworkRegime.java

```diff
@@ -0,0 +1,87 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Validates that pods do not move from a virtual network that does not use port mapping (and thus ignores the ports
+ *  as Mesos resources) to a network (virtual or Host) that does use the ports on the host machine.
+ */
+public class PodSpecsCannotChangeNetworkRegime implements ConfigValidator<ServiceSpec> {
+    @Override
+    public Collection<ConfigValidationError> validate(ServiceSpec nullableOldConfig, ServiceSpec newConfig) {
+        Pair<List<ConfigValidationError>, Map<String, PodSpec>> pair = validateInitialConfigs(
+                nullableOldConfig, newConfig);
+        List<ConfigValidationError> errors = pair.a;
+        Map<String, PodSpec> newPods = pair.b;
+        if (newPods.isEmpty()) {
+            return errors;
+        }
+
+        // check the PodSpecs to make sure none of them make a transition form a state where they use host ports
+        // to one where they don't (or vice versa).
+        Map<String, PodSpec> oldPods = nullableOldConfig.getPods().stream()
+                .collect(Collectors.toMap(PodSpec::getType, podSpec -> podSpec));
+        for (Map.Entry<String, PodSpec> kv: newPods.entrySet()) {
+            PodSpec newPod = kv.getValue();
+            // first we check that the new pod's networks (if present) do not erroneously use port mapping
+            List<String> offendingNetworkNames = getNetworksWithInvalidPortMappings(newPod);
+            if (offendingNetworkNames.size() > 0) {
+                errors.add(ConfigValidationError.transitionError(
+                        String.format("PodSpec[name:%s]", newPod.getType()),
+                        "null",
+                        newPod.getNetworks().toString(),
+                        String.format("New config has pod %s that indicates port mapping for virtual network(s) %s, " +
+                                        "that do not support port mapping.", newPod.getType(),
+                                StringUtils.join(offendingNetworkNames, ", "))));
+            }
+            // now we check that the none of the new pods move from not using host ports to using them (or vice
+            // versa)
+            PodSpec oldPod = oldPods.get(kv.getKey());
+            if (oldPod != null && podSpecUsesHostPorts(oldPod) != podSpecUsesHostPorts(newPod)) {
+                errors.add(ConfigValidationError.transitionError(
+                        String.format("PodSpec[name:%s]", oldPod.getType()),
+                        oldPod.getNetworks().toString(),
+                        newPod.getNetworks().toString(),
+                        String.format("New config has pod %s moving networks from %s to %s, changing it's " +
+                                        "host ports requirements from %s to %s, not allowed.",
+                                newPod.getType(), oldPod.getNetworks().toString(), newPod.getNetworks().toString(),
```

Comment:
For future reference, you can omit the `.toString()`:
`If the argument is null, then the result is "null". If the argument implements Formattable, then its formatTo method is invoked. Otherwise, the result is obtained by invoking the argument's toString() method.`

Also note: this protects you against throwing NPEs in your logging statements (ask me how I know)